client/src/legacy/LeftAndMain.js

```diff
@@ -643,6 +640,33 @@ $.entwine('ss', function($) {
       return promise;
     },
 
+    /**
+     * Reverts the window.history state back to the last known good state. See this.handleStateChange().
+     */
+    reverseStateChange: function() {
```

Comment:
do we also need to call `this.setStateChangeCount(this.getStateChangeCount() + 1);`, or `this.setStateChangeCount(this.getStateChangeCount() - 1);` ?

Reply:
I only abstracted [this block of code](https://github.com/silverstripe/silverstripe-admin/pull/1280/files#diff-3f04cea4046f53f0d7ea15fb030d44ab684baf2290fa998b5e2462feb8fe5fd8L575-L587) to a separate method (and ideally it'd be private/protected) so that it could be called multiple times within the legacy `handleStateChange` method. So, no I don't think so. 

Maybe I should consider declaring it as a function/closure inside of the `handleStateChange` method?

Reply:
Oh rats, sorry I misread your comment. Yeah that's probably a good idea.

Reply:
Sorry for the delay on this @emteknetnz.

Went ahead and pushed an extra line to also decrement that `StateChangeCount` property, see e129d254e728f6a7fcf3dca75b315286ab88debf. Good idea to keep the API consistent (albeit I don't see it referenced anywhere _within_ the module). Best not to break anything referencing it for a bug fix.